lib/preneeds/service.rb

```diff
@@ -63,5 +87,46 @@ def receive_pre_need_application(params)
     def savon_client
       @savon ||= Savon.client(wsdl: Settings.preneeds.wsdl)
     end
+
+    def make_net_http_request(body, headers)
+      uri = URI(config.base_path)
+
+      Net::HTTP.start(uri.host, uri.port) do |http|
+        begin
+          http.use_ssl = (uri.scheme == 'https')
+          request = Net::HTTP::Post.new(uri.path, headers)
+          request.body = body
+          http.request(request)
+        rescue Timeout::Error, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+               Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError # => e
+          # TODO: log_sentry and have a manual strategy in place for resolving this maybe.
+          # TODO: trigger breakers
+          {}
+        end
+      end
+    end
+
+    def build_multipart(soap, attachment)
+      multipart = Mail.new do
+        content_type 'multipart/related; type="application/xop+xml"'
+        content_transfer_encoding 'chunked'
+      end
+
+      soap_part = Mail::Part.new do
+        content_type 'text/xml; charset="utf-8"'
+        body soap.body
+      end
+
+      multipart.add_part(soap_part)
+
+      multipart.attachments[attachment.file.original_filename] = {
+        mime_type: 'application/pdf',
```

Comment:
well the only files that the host allows is 'application/pdf' if they provide anything that doesn't match it should be an error, I suppose we could do a file type validation on the attachment model if that's what you're getting at?